### PR TITLE
Separate fb code

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
This pull request just separates the Facebook code from the server code, and makes no adjustments to the Facebook code as of #296. 

#296 should be merged/rejected before this PR is considered.

### CODE REVIEW

File: hs/CourseographyFacebook.hs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+-- | Posts a message to Facebook.
+postToFB :: (MonadResource m, MonadBaseControl IO m) => String -> FB.UserAccessToken -> FB.FacebookT FB.Auth m FB.Id
+postToFB code token = FB.postObject "me/feed" [args2] token
+
+-- | Gets a users Facebook email.
+getEmail :: String -> ServerPart Response
+getEmail code = liftIO $ retrieveFBData code
+
+fbAuth1Url :: String
```
Comment (david-yz-liu) [inline hs/CourseographyFacebook.hs]: Can you move all URLs to the top, and factor out the domain `localhost:8000` into a separate value?

File: hs/CourseographyFacebook.hs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+fbAuth1Url = "https://www.facebook.com/dialog/oauth?client_id=442286309258193&redirect_uri=http://localhost:8000/test&scope=user_birthday"
```
Comment (david-yz-liu) [inline hs/CourseographyFacebook.hs]: And along a similar line, put the app id into a separate value as well.

File: hs/CourseographyFacebook.hs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+-- | Gets a user access token.
+getToken :: (MonadResource m, MonadBaseControl IO m) => FB.RedirectUrl -> String -> FB.FacebookT FB.Auth m FB.UserAccessToken
+getToken url code = FB.getUserAccessTokenStep2 url [args code]
+
+-- | Posts a message to Facebook.
+postToFB :: (MonadResource m, MonadBaseControl IO m) => String -> FB.UserAccessToken -> FB.FacebookT FB.Auth m FB.Id
+postToFB code token = FB.postObject "me/feed" [args2] token
+
+-- | Gets a users Facebook email.
```
Comment (david-yz-liu) [inline hs/CourseographyFacebook.hs]: Missing apostrophe

File: hs/CourseographyFacebook.hs (hunk 2 above)
Comment (Ian-Stewart-Binks) [inline hs/CourseographyFacebook.hs]: Ah, this has been changed in a future commit that is not in this pull request. I will soon create a pull request with some cleaner Facebook code.

File: hs/CourseographyFacebook.hs (hunk 1 above)
Comment (Ian-Stewart-Binks) [inline hs/CourseographyFacebook.hs]: Will do.

File: hs/CourseographyFacebook.hs (hunk 2 above)
Comment (david-yz-liu) [inline hs/CourseographyFacebook.hs]: If it's just code cleanup, you can put the commits in with this one.

**Pull request merged**